File: WdtcCore/src/main/java/org/wdt/wdtc/download/DownloadGameVersion.java
package org.wdt.wdtc.download;

import org.wdt.wdtc.download.game.DownloadVersionGameFile;
import org.wdt.wdtc.game.Launcher;

import java.io.IOException;

public class DownloadGameVersion {
    protected final Launcher launcher;
    protected final DownloadVersionGameFile DownloadGame;
    protected final boolean install;

    public DownloadGameVersion(Launcher launcher) {
        this(launcher, false);
    }

    public DownloadGameVersion(Launcher launcher, boolean Install) {
        this.launcher = launcher;
        this.install = Install;
        this.DownloadGame = new DownloadVersionGameFile(launcher, Install);
    }

    public void DownloadGameFile() throws IOException {
        DownloadGame.DownloadGameVersionJson();
        DownloadGame.DownloadGameAssetsListJson();
        DownloadGame.DownloadVersionJar();
    }

    public void DownloadGameLibrary() {
        DownloadGame.DownloadGameLibraryFileTask().DownloadLibraryFile();
    }

    public void DownloadResourceFile() {
        DownloadGame.getDownloadGameAssetsFile().DownloadAssetsFiles();
    }

    public void DownloadGame() throws IOException {
        DownloadGameFile();
        DownloadGameLibrary();
        DownloadResourceFile();
    }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/download/DownloadGameVersion.kt
package org.wdt.wdtc.core.download

import org.wdt.wdtc.core.download.game.DownloadVersionGameFile
import org.wdt.wdtc.core.game.*
import java.io.IOException

open class DownloadGameVersion @JvmOverloads constructor(
    protected val launcher: Launcher,
    protected val install: Boolean = false
) {
  protected val downloadGame: DownloadVersionGameFile = DownloadVersionGameFile(launcher, install)

  @Throws(IOException::class)
  fun downloadGameFileTask() {
    downloadGame.startDownloadGameVersionJson()
    downloadGame.startDownloadGameAssetsListJson()
    downloadGame.startDownloadVersionJar()
  }

  fun downloadGameLibraryTask() {
    downloadGame.downloadGameLibraryFileTask.startDownloadLibraryFile()
  }

  fun downloadResourceFileTask() {
    downloadGame.downloadGameAssetsFile.startDownloadAssetsFiles()
  }

  @Throws(IOException::class)
  fun startDownloadGame() {
    downloadGameFileTask()
    downloadGameLibraryTask()
    downloadResourceFileTask()
  }
}
